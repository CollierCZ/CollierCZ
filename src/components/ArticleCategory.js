import React from "react";
import { navigate } from "gatsby";
import { string } from "prop-types";
import { slugify } from "../utilities/CaseHelpers";
import { TextLink } from "@kiwicom/orbit-components";

const ArticleCategory = ({ prefix, category }) => {
  return (
    <span>
      {prefix}
      <span key={category}>
        <TextLink
          href={`/category/${slugify(category)}`}
          type="secondary"
          onClick={(event) => {
            event.preventDefault();
            navigate(`/category/${slugify(category)}`);
          }}
          stopPropagation
        >
          {category}
        </TextLink>
      </span>
    </span>
  );
};

ArticleCategory.propTypes = {
  prefix: string,
  category: string.isRequired,
};

export default ArticleCategory;
