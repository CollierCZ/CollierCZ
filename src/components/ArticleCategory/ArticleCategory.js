import React, { Component } from "react";
import _ from "lodash";
import { Link } from "gatsby";
import "./ArticleCategory.css";

class ArticleCategory extends Component {
  render() {
    const { prefix, category } = this.props;
    if (category) {
      return (
                <span>
          {prefix}
          <span key={category}>
            <Link key={category} to={`/category/${_.kebabCase(category)}`}>
              {category}
            </Link>
          </span>
        </span>
      );
    }
    return null;
  }
}

export default ArticleCategory;
