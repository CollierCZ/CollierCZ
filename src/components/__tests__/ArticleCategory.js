import React from "react"
import renderer from "react-test-renderer"
import ArticleCategory from "../ArticleCategory"

describe("ArticleCategory", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ArticleCategory category="Education" prefix=" in " />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})