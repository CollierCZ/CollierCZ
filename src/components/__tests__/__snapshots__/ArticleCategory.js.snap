// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCategory renders correctly 1`] = `
.c0 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-weight: 500;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
}

.c0,
.c0:link,
.c0:visited {
  color: #252A31;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c0:hover,
.c0:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #00A991;
}

.c0:hover .TextLink__IconContainer-sc-1bvlje4-0,
.c0:active .TextLink__IconContainer-sc-1bvlje4-0 {
  color: #00A991;
}

.c0:focus {
  outline: none;
  background: rgba(95,115,140,0.1);
  box-shadow: 0 0 0 4px rgba(95,115,140,0.1);
}

.c0:focus:not(:focus-visible) {
  background: rgba(95,115,140,0.1);
  box-shadow: 0 0 0 4px rgba(95,115,140,0.1);
}

.c0:-moz-focusring,
.c0:focus-visible {
  background: rgba(95,115,140,0.1);
  box-shadow: 0 0 0 4px rgba(95,115,140,0.1);
}

<span>
   in 
  <span>
    <a
      className="c0"
      href="/category/education"
      onClick={[Function]}
      rel=""
    >
      Education
    </a>
  </span>
</span>
`;
